include/cell/traits/copy.hpp

```diff
@@ -0,0 +1,88 @@
+          typename Base = TraitsBase<Element_>>
+struct G2S2DCopyTraits : public Base {
+    using Element = Element_;
+
+    static constexpr int kRows = kRows_;
+    static constexpr int kCols = kCols_;
+
+    static constexpr int kShmRows = kShmRows_;
+    static constexpr int kShmCols = kShmCols_;
+
+    using SrcLayout = RowMajor<kRows, kCols, kCols>;
+
+    // To avoid bank conflicts, the shared memory requires a swizzled layout
+    static constexpr int kSwizzleMode = kShmCols % 32 ? 1 : 0;
+    using Swizzled =
+        SwizzledRowMajor<Element, kShmRows, kShmCols, kSwizzleMode>;
+    using DstLayout = typename Swizzled::SmemLayout;
+
+    // threads in a thread block are laid out as a 2D tile
+    // that has a shape of kThreadsRows x kThreadsCols.
+    static constexpr int kThreadsCols = kShmCols / Base::kNumPerAccess;
+    static constexpr int kThreadsRows = kThreads / kThreadsCols;
+    using ThreadLayout = RowMajor<kThreadsRows, kThreadsCols, kThreadsCols>;
+
+    using ValueLayout = Layout<Shape<_1, Int<Base::kNumPerAccess>>>;
+
+#if defined(CUTE_ARCH_CP_ASYNC_SM80_ENABLED)
+    using CopyInst =
+        Copy_Atom<SM80_CP_ASYNC_CACHEGLOBAL<cute::uint128_t>, Element>;
+#else
+    using CopyInst = Copy_Atom<DefaultCopy, Element>;
+#endif
+
+    using TiledCopy =
+        decltype(make_tiled_copy(CopyInst{}, ThreadLayout{}, ValueLayout{}));
+};
+
+/// @brief Configurations for layout and shape of the 2D data tile.
+/// @tparam Element_: the element type
+/// @tparam kThreads
+template <typename Element_, const int kRows_, const int kCols_,
+          const int kShmRows_, const int kShmCols_, const int kThreads,
+          typename Base = TraitsBase<Element_>>
+struct S2G2DCopyTraits : public Base {
+    using Element = Element_;
+
+    static constexpr int kRows = kRows_;
+    static constexpr int kCols = kCols_;
+
+    static constexpr int kShmRows = kShmRows_;
+    static constexpr int kShmCols = kShmCols_;
+
+    static constexpr int kSwizzleMode = kShmCols % 32 ? 1 : 0;
+    using Swizzled =
+        SwizzledRowMajor<Element, kShmRows, kShmCols, kSwizzleMode>;
+    using SrcLayout = typename Swizzled::SmemLayout;
+
+    // To avoid bank conflicts, the shared memory requires a swizzled layout
+    using DstLayout = RowMajor<kRows, kCols, kCols>;
+
+    // threads in a thread block are laid out as a 2D tile
+    // that has a shape of kThreadsRows x kThreadsCols.
+    static constexpr int kThreadsCols = kShmCols / Base::kNumPerAccess;
+    static constexpr int kThreadsRows = kThreads / kThreadsCols;
+    using ThreadLayout = RowMajor<kThreadsRows, kThreadsCols, kThreadsCols>;
+
+    using ValueLayout = Layout<Shape<_1, Int<Base::kNumPerAccess>>>;
+
+    // transfer data from global memory to shared memory has cp.async,
+    // while transfer data from shared memory to global memory does not have.
+    // for the latter case, the copy instruction should be the default one.
+    using TiledCopy = decltype(make_tiled_copy(
```

Comment:
I have attempted to unify the implementation of data transfer between shared memory and global memory.

During this process, I realized that the only key difference is the copy instruction used for this process. Specifically, transferring data tiles from global to shared memory utilizes `cp.async` on Ampere, whereas the reverse process does not.